test/test_resolve_links.cpp

```diff
@@ -131,6 +133,48 @@ TORRENT_TEST(resolve_links)
 		TEST_EQUAL(num_matches, e.expected_matches);
 
 	}
-#endif // TORRENT_DISABLE_MUTABLE_TORRENTS
 }
 
+// this ensure that internally the is a range lookup
+// since the zero-hash piece is in the second place
+TORRENT_TEST(range_lookup_duplicated_files)
+{
+	file_storage fs1;
+	file_storage fs2;
+
+	fs1.add_file("test_resolve_links_dir/tmp1", 1024);
+	fs1.add_file("test_resolve_links_dir/tmp2", 1024);
+	fs2.add_file("test_resolve_links_dir/tmp1", 1024);
+	fs2.add_file("test_resolve_links_dir/tmp2", 1024);
+
+	libtorrent::create_torrent t1(fs1, 1024);
+	libtorrent::create_torrent t2(fs2, 1024);
+
+	t1.set_hash(0, sha1_hash::max());
```

Comment:
the hashes by default are `0` (an internal `m_piece_hash.resize(m_files.num_pieces())`), I only set the `t1` first piece hash to `max` and that do the trick of having the same size but different hash compared to the `t2`.